### PR TITLE
Feature/5713 add hover tip for spatial extents

### PR DESCRIPTION
**Pull request by NekoLyn**
1. fix showing overlapping spatial extents when click on a single unclustered point
2. create a simple popup for hovering on a single unclustered point (different from design which is too big and contains too much information, and may need change later according to Steve @quye04 )
3. adjust search page height to be adaptive to the screen height 
4. adjust the cluster size to make less unclustered points on map

### CODE REVIEW

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:218]: rename this, too clunky, e.g `onUnclusterLeave`

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:226]: rename this too

e.g `onClusterEnter`

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:233]: rename this too . e.g `onClusterLeave`

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:240]: rename this , e.g `onUnclusterClick`

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:460]: line 256 of this file, rename too , e.g `clusterLayerMouseClickEventHandler` to `onClusterClick`

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:402]: consider suggested grouped arbitrary constants in previous comments

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:413]: consider suggested grouped arbitrary constants in previous comments

**Comment by vietnguyengit** [comment]
I'm reviewing it's a long one @utas-raymondng , FYI so you won't do duplicate work.

@NekoLyn if you requested me and I'm in it, make sure to communicate before you request another person.

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:358]: catch error then log it too, if you simply ignore the exception, then log something like : <...> skipped ... blah blah

File: src/components/map/mapbox/popup/MapPopup.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/popup/MapPopup.tsx:100]: if this component be regularly rendered, not a requested change, but you can consider this, optionally, 
https://react.dev/reference/react/memo

https://react.dev/reference/react/memo#should-you-add-memo-everywhere

**Review by vietnguyengit — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/as-a--type-of-user--i-want--some-goal--so-that--some-reason-.md
Comment (utas-raymondng) [reply .github/ISSUE_TEMPLATE/as-a--type-of-user--i-want--some-goal--so-that--some-reason-.md:1]: it is code merged, my bad on this one that somehow it goes to the same branch, so please ignore this one

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (utas-raymondng) [reply src/components/map/mapbox/layers/ClusterLayer.tsx:133]: I am creating an ErrorBoundary component in #5459 which should catch this and display an ugly error page

File: src/components/map/mapbox/popup/MapPopup.tsx
Comment (NekoLyn) [reply src/components/map/mapbox/popup/MapPopup.tsx:57]: OK. This is the previous design that popup contains the list result card however it contains much unnecessary information and Steve @quye04  will re-design the popup card. I will remove this part.

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (utas-raymondng) [reply src/components/map/mapbox/layers/ClusterLayer.tsx:61]: these contants is very specific to this layer so it makes sense to put it there but grouping it is good to make it more readable, it will be good to have those to be default props of the component so user can change it via props

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (utas-raymondng) [reply src/components/map/mapbox/layers/ClusterLayer.tsx:507]: Yes it is a strange behavior that if getSource failed aka the source do not exist it throw exception.

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (NekoLyn) [reply src/components/map/mapbox/layers/ClusterLayer.tsx:133]: Thanks! I commented a TODO task here

**Comment by utas-raymondng** [comment]
@NekoLyn remove me from reviewer, I let @vietnguyengit do it this time

File: src/components/map/mapbox/popup/MapPopup.tsx
Comment (NekoLyn) [reply src/components/map/mapbox/popup/MapPopup.tsx:100]: This component has different props for each render when hover on different points, so may not proper for using React.memo. Conditional rendering is added for this component to avoid unccecessary rendering: if isLoading it will early return a loading status component; in clusterLayer it wait for get collection to trigger the MapPopup render.

**Comment by NekoLyn** [comment]
> @NekoLyn remove me from reviewer, I let @vietnguyengit do it this time

somehow cannot remove you from reviewer since you have made comments...I think one approvement is enough maybe..If it needs your approvement I will ping you

**Comment by vietnguyengit** [comment]
I saw you've ticked "Resolved" for the comments but I cannot find new commits or new changes to some comments requiring code changes @NekoLyn

**Comment by NekoLyn** [comment]
> I saw you've ticked "Resolved" for the comments but I cannot find new commits or new changes to some comments requiring code changes @NekoLyn

Hi I just request the review. Had some problem with pc just restarted it and commit...

**Review by utas-raymondng — APPROVED** [review]
I leave this to @vietnguyengit

File: src/components/map/mapbox/layers/ClusterLayer.tsx
Comment (vietnguyengit) [inline src/components/map/mapbox/layers/ClusterLayer.tsx:196]: inline return possible here, `return collections.collections[0];`

Comment (vietnguyengit) [reply src/components/map/mapbox/layers/ClusterLayer.tsx:196]: last minor change @NekoLyn

**Review by vietnguyengit — APPROVED** [review]

**Comment by vietnguyengit** [comment]
just wonder why your commits always have `main` merged line ? @NekoLyn

**Comment by vietnguyengit** [comment]
was that git asking you to rebase  or you choose to do so? @NekoLyn

**Comment by NekoLyn** [comment]
> was that git asking you to rebase or you choose to do so? @NekoLyn

I choose to do it. but githut will also detect if the commit is ahead or behind 'main'. if the commits are ahead of 'main', then someone can merge the pr. however if the commits are behind 'main' then you need to rebase before merge, which may casue unsolved conflicts. That's why i will check any updates of 'main' and merge then resolve potential confilcts before pushing commit.

**Comment by NekoLyn** [comment]
> just wonder why your commits always have `main` merged line ? @NekoLyn

This line is I merge lastest pr that mereged to 'main'. The commit message is different from Raymond and Havier due to using different IDE (I am using VS code and they may use the Intellij)

**Pull request merged**
